app/helpers/validation/profile.py

```diff
@@ -20,7 +20,7 @@ def read_linestring(request_object):
     if 'geom' in request_object.args:
         linestring = request_object.args.get('geom')
     elif request_object.is_json and len(request_object.get_data(as_text=True)) > 0:
-        linestring = request_object.get_json()
+        linestring = request_object.get_data(as_text=True)  # read as text
```

Comment:
We should consider also checking the 'Content-Length' header before reading the payload to avoid memory starvation. I don't know what would be the biggest payload required, but I guess that having a limit at 32MB is safe to avoid starvation and accept all use case ?

Reply:
This would reduce the load on the server (is_json reads the content type header if I'm not mistaken) by reading the headers rather than the content as its first step. 
I'll do that tomorrow morning, first thing.